src/main/java/com/xceptance/xlt/engine/httprequest/HttpRequest.java

```diff
@@ -362,22 +362,41 @@ public HttpRequest removeHeaders()
      */
     public HttpRequest param(final String name, String value)
     {
+        param(new NameValuePair(name, value));
+
+        return this;
+    }
+
+    /**
+     * Adds a request parameter with the given name/value pair.
+     *
+     * @param nameValuePair
+     *            the name/value pair representing the parameter to add
+     */
+    public HttpRequest param(NameValuePair nameValuePair)
+    {
+        String name = nameValuePair.getName();
+
         if (StringUtils.isBlank(name))
         {
             throw new IllegalArgumentException("Parameter name must not be blank.");
```

Comment:
Message is a little bit confusing as there is no 'name' parameter. Hence, I'd change to something like "NameValuePair parameter is invalid since its name is blank".

Reply:
Hm, difficult. The message must fit for all overloads, also for those that do not deal with NameValuePair directly. You know? And if you read the message as "Parametername ..." instead of "Parameter 'name' ...", the message will sound much better, eh? ;-)

Reply:
Tricky, indeed. What about to switch the first two words: "Name of parameter must not be blank"?

Reply:
Done. Rephrased the other message in this method accordingly.